text/0000-monorepo.md

```diff
@@ -0,0 +1,359 @@
+  
+This strategy bleeds into issue templates in that we would most likely want a specific structure around issue titles and contents for things like bugs, feature requests, and support.
+
+Pull request templates would ideally follow our current format where they have a corresponding issue and provide a high-level change log and steps to review.
+
+# Drawbacks
+
+The main disadvantages of using the strategies outlined above is that it forces our model into a monorepo. While this in and of itself is not a negative thing, it does shift a portion of our development time towards investing in tooling to help make working in a monorepo more enjoyable.
+
+Aspects of this include a custom CLI to help with working on a single component, in addition to tooling or strategies for scaling out our core tools so that they are either fast on a large number of files, or only run on files that have chanegd.
+
+In addition, while moving work to one project eliminates any confusion to issue creation, we run into a problem where everyone will get additional notifications that may not be related to their core project. The immediate action after this would be to ignore conversations that one is not actively participating in, but this could lead to issues not getting any response.
+
+There are ways to mitigate this, including issue templates, but ultimately scaling issues and pull requests in a monorepo with multiple teams/collaboratos is still an unknown.
+
+Finally, there is a concern around making everything under the Carbon umbrella public. The main use-case being: what if a team wants to use Carbon to build a component but is unable to make this component open-source? One solution could be the mirror our project to an internal GitHub Enterprise repo, but this could cause fragmentation in the ecosystem.
+
+# Alternatives
+
+The main alternative would be for **only** Carbon core to have a monorepo design. Given a desire to want to ship per-component packages, it is hard to see how we could scale doing repo-per-compnoent efforts. Particularly with respect to consistency between projects, keeping things up-to-date, and the experience of working with interlinked components, aka compound components.
+
+# Adoption strategy
+
+The majority of this work would be internal to the Carbon core team, or for the maintainers of specific frameworks and add-ons.
+
+The ideal would be to break up this proposal into several stages that can then help determine a ship/no-ship status for this approach. These stages include:
+
+- Stage 1: Convert existing carbon core projects to a single monorepo (the Carbon repo)
+- Stage 2: Add framework-specific projects to the Carbon repo
+- Stage 3: Add add-on components to the Carbon repo
+
+Alongside these stages, we would need to define several items including:
+
+- What is the process for adding a new component package? Is it okay if this package lives only for a particular framework?
+- What is the release cadence like for packages?
```

Comment:
My suggestion:

Major - TBD, but our current cadence is once every 6 months?
Minor, Patch - automatic